util.go

```diff
@@ -170,3 +171,29 @@ func Equal(a, b interface{}) bool {
 	}
 	return false
 }
+
+// Walk is a filepath.Walk helper which follows symlinks and always returns the
+// path of the file or directory that is referenced in the link.
+func Walk(root string, walkFn filepath.WalkFunc) error {
```

Comment:
I found this patch while searching for symlink aware dir traversal. Thank you.
There are a few issues with this: I implemented an alternative code here: https://gist.github.com/pavanka/e22a432fd6487fc8e578